test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestFixture.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public abstract class EndToEndTestFixture : IDisposable
+    {
+        private string _copiedRootPath;
+
+        protected EndToEndTestFixture(string rootPath, string testId)
+        {
+            FixtureId = testId;
+            string connectionString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(connectionString);
+
+            QueueClient = storageAccount.CreateCloudQueueClient();
+            BlobClient = storageAccount.CreateCloudBlobClient();
+            TableClient = storageAccount.CreateCloudTableClient();
+
+            CreateTestStorageEntities().Wait();
+
+            _copiedRootPath = Path.Combine(Path.GetTempPath(), "FunctionsE2E", DateTime.UtcNow.ToString("yyMMdd-HHmmss"));
+            FileUtility.CopyDirectory(rootPath, _copiedRootPath);
+
+            // Allow derived classes to limit functions. We'll update host.json in the copied location
+            // so it only affects this fixture's instance.
+            IEnumerable<string> functions = GetActiveFunctions();
+            string hostJsonPath = Path.Combine(_copiedRootPath, "host.json");
+            JObject hostJson = JObject.Parse(File.ReadAllText(hostJsonPath));
+            if (functions != null && functions.Any())
+            {
+                hostJson["functions"] = JArray.FromObject(functions);
+            }
+            File.WriteAllText(hostJsonPath, hostJson.ToString());
+
+            Host = new TestFunctionHost(_copiedRootPath);
+            Host.SetNugetPackageSources("http://www.myget.org/F/azure-appservice/api/v2", "https://api.nuget.org/v3/index.json");
+            Host.InstallBindingExtension("Microsoft.Azure.WebJobs.Extensions.CosmosDB", "3.0.0-beta7-10595").Wait();
```

Comment:
In the interest of time, this is hardcoded right now. We want to have this pull from some configuration in the future to allow us to target different package versions easily.